src/analysis/play-utilities/PlayDecider.cpp

```diff
@@ -6,5 +6,18 @@
 
 namespace rtt::ai::analysis {
 // Maybe plays should have some sort of score based on the world how good they are
-Play PlayDecider::decideBestPlay(world::World *world, world::Field *field, std::vector<Play> validPlays) { return Play(); }
-}  // namespace rtt::ai::analysis
\ No newline at end of file
+    std::shared_ptr<Play> PlayDecider::decideBestPlay(world::World *world, world::Field *field, std::vector<std::shared_ptr<Play>> validPlays) {
+        int max = -1;
+        if (validPlays.empty()) {
+            std::cerr << "valid plays are empty" << std::endl;
+        }
+        for (auto const& play : validPlays) {
+            int temp = play->scorePlay(world, field);
+            if (temp > max) {
+                max = temp;
+            }
+            bestPlay = play;
+        }
```

Comment:
```cpp
if (validPlays.empty()) {
    std::cerr << "Valid plays are empty\n";
    return;
}

return *std::max_element(validPlays.begin(), validPlays.end(), [&](auto& play) {
    return plsy->scorePlay(world, field);
});
```

I personally think this is cleaner, and makes more sense, std::max_element, who doesn't understand that